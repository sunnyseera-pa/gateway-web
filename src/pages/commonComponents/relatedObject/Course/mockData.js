export default {
    _id: '5f96f3450d236c6cafdaa473',
    domains: ['test'],
    award: ['BSc'],
    courseOptions: [
        {
            flexibleDates: false,
            _id: '5f96f3450d236c6cafdaa476',
            startDate: '2020-11-12T00:00:00.000Z',
            studyMode: 'Full-time',
            studyDurationNumber: 5,
            studyDurationMeasure: 'Day(s)',
            fees: [
                {
                    _id: '5f96f3450d236c6cafdaa477',
                    feeDescription: 'Test',
                    feeAmount: 4,
                    feePer: 'Month',
                },
            ],
        },
        {
            flexibleDates: false,
            _id: '5f96f3450d236c6cafdaa476',
            startDate: '2020-11-12T00:00:00.000Z',
            studyMode: 'Full-time',
            studyDurationNumber: 5,
            studyDurationMeasure: 'Day(s)',
            fees: [
                {
                    _id: '5f96f3450d236c6cafdaa477',
                    feeDescription: 'Test',
                    feeAmount: 4,
                    feePer: 'Month',
                },
            ],
        },
    ],
    id: 3491286126459112,
    type: 'course',
    activeflag: 'active',
    title: 'Test A course',
    provider: 'Test A',
    description: 'Test A Test A Test A edited Test A',
    counter: 208,
    relatedresources: 3,
    updated: 1621465200000,
};
