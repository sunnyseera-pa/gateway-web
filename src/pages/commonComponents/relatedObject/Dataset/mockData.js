export default {
    categories: {
        programmingLanguage: [],
    },
    tags: {
        features: ['COVID-19', 'TRACKER', 'SAIL', 'COVID', 'MOBILE APP', 'CORONAVIRUS', 'SYMPTOM', 'NCS', 'National Core Study'],
        topics: [],
    },
    datasetfields: {
        publisher: 'ALLIANCE > TestSAIL',
        geographicCoverage: ['Great Britain'],
        physicalSampleAvailability: ['NOT AVAILABLE'],
        abstract: 'This is a test dataset abstract',
        ageBand: '15-150',
        metadataquality: {
            quality_score: 89.48,
        },
        phenotypes: [],
    },
    programmingLanguage: [],
    pid: 'c1f4b16c-9dfa-48e5-94ee-f0aa58c270e4',
    id: 401490453939233,
    datasetid: '78ce070c-84db-4676-81c5-0c672f4e12e3',
    type: 'dataset',
    activeflag: 'active',
    name: 'COVID-19 Symptom Tracker Dataset',
    description: 'This is a test dataset description',
    license: 'https://covid.joinzoe.com/privacy-notice',
    datasetv2: {
        summary: {
            publisher: {
                name: 'sail',
                logo: '',
                memberOf: 'ALLIANCE',
            },
        },
    },
    counter: 955,
    isCohortDiscovery: false,
    persons: [
        {
            firstname: 'Rich',
            lastname: 'Example',
        },
    ],
    latestUpdate: '2021-05-04T14:14:49.389Z',
    relatedresources: 1,
};
