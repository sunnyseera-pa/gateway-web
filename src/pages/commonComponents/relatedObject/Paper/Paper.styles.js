import { css } from '@emotion/react';

export const author = css`
	margin-top: 2px;
`;

export const journal = css`
	margin-top: 10px;
`;
