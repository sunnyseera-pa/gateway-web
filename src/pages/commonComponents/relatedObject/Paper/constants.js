const tab_url = '/search?search=&tab=Paper';
export const paper = {
    TAB: 'Paper',
    FEATURES: {
        parentKey: 'paperfeatures',
        filter: 'paperFeaturesSelected',
        tagType: 'tag',
        url: `${tab_url}&paperfeatures=`,
    },
    TOPICS: {
        parentKey: 'papertopics',
        filter: 'paperTopicsSelected',
        tagType: 'tag',
        url: `${tab_url}&papertopics=`,
    },
};
