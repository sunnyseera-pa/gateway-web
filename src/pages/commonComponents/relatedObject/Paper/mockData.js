export default {
    _id: '5f156b42577505d15e88f239',
    categories: {
        programmingLanguage: [],
    },
    tags: {
        features: ['Arbitrage'],
        topics: ['Blood'],
    },
    authors: [947228017269611],
    id: 18656838486747996,
    type: 'paper',
    name: 'This is a test paper',
    description: 'test paper description',
    activeflag: 'active',
    programmingLanguage: [],
    counter: 68,
    relatedresources: 5,
    persons: [
        {
            _id: '5e6f984a0a7300dc8f6fb196',
            id: 947228017269611,
            firstname: 'Test',
            lastname: 'Person',
        },
    ],
    updated: 1634511600000,
};
