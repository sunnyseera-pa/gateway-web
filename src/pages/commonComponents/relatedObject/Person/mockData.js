export default {
	_id: '5e544facbd427b6e9cd9059e',
	id: 900000011,
	lastname: 'Nei',
	bio: 'Jefferson Comprehensive Epilepsy Center',
	firstname: 'Maromi',
	type: 'person',
	tags: ['Neuroendocrinology', 'Epilepsy', 'Neuroscience'],
	activeflag: 'active',
	authors: [''],
	counter: 73,
	relatedresources: 0,
	persons: [],
	updated: 1621551600000,
};
