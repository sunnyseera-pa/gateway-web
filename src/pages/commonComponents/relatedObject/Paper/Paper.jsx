/** @jsx jsx */
import { jsx } from '@emotion/react';
import React from 'react';
import { Row, Col } from 'react-bootstrap';
import PropTypes from 'prop-types';
import { stripMarkdown } from '../../../../utils/GeneralHelper.util';
import SVGIcon from '../../../../images/SVGIcon';
import RemoveButton from '../RemoveButton/RemoveButton';
import Title from '../Title/Title';
import Description from '../Description/Description';
import Tag from '../Tag/Tag';
import { ReactComponent as LockSVG } from '../../../../images/icon-security.svg';
import { paper } from './constants';
import * as styles from './Paper.styles';
import '../../CommonComponents.scss';

const Paper = ({ data, activeLink, onSearchPage, showRelationshipQuestion, updateOnFilterBadge, removeButton, isLocked, onClick }) => (
    <Row data-test-id='related-paper-object' className='noMargin'>
        <Col sm={10} lg={10} className='pad-left-24'>
            <Title
                activeLink={activeLink}
                name={data.name}
                id={data.id}
                type='paper'
                onClickHandler={() => {
                    onClick();
                }}
            />
            <div className='gray800-14' css={styles.author}>
                {data.authorsNew}
            </div>

            <div className='gray800-14' css={styles.journal}>
                {data.journal} {data.journalYear}
            </div>
        </Col>
        <Col sm={2} lg={2} className={isLocked ? 'lockSVG pad-right-24' : 'pad-right-24'}>
            {showRelationshipQuestion ? isLocked ? <LockSVG /> : <RemoveButton removeButtonHandler={removeButton} /> : ''}
        </Col>
        <Col sm={12} lg={12} className='pad-left-24 pad-right-24 pad-top-16'>
            <Tag tagName={paper.TAB} tagType={data.type} updateOnFilterBadgeHandler={updateOnFilterBadge}>
                <SVGIcon name='newprojecticon' fill='#3c3c3b' className='badgeSvg mr-2' viewBox='-2 -2 22 22' />
            </Tag>
            {data.tags.features &&
                data.tags.features.map((feature, index) => (
                    <Tag
                        key={`${feature}-${index}`}
                        tagName={feature}
                        activeLink={activeLink}
                        onSearchPage={onSearchPage}
                        updateOnFilterBadgeHandler={updateOnFilterBadge}
                        {...paper.FEATURES}
                    />
                ))}

            {data.tags.topics &&
                data.tags.topics.map((topic, index) => (
                    <Tag
                        key={`${topic}-${index}`}
                        tagName={topic}
                        activeLink={activeLink}
                        onSearchPage={onSearchPage}
                        updateOnFilterBadgeHandler={updateOnFilterBadge}
                        {...paper.TOPICS}
                    />
                ))}
        </Col>
        {!showRelationshipQuestion && <Description type={data.type} description={stripMarkdown(data.description, 255)} />}
    </Row>
);

Paper.propTypes = {
    data: PropTypes.object.isRequired,
    activeLink: PropTypes.bool.isRequired,
    showRelationshipQuestion: PropTypes.bool.isRequired,
    onSearchPage: PropTypes.bool.isRequired,
    updateOnFilterBadge: PropTypes.func.isRequired,
    removeButton: PropTypes.func.isRequired,
};

Paper.defaultProps = {
    onClick: () => {},
};

export default Paper;
