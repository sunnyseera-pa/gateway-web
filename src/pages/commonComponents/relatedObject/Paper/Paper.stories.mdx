import { Meta, Story, Canvas } from '@storybook/addon-docs';
import Paper from './Paper';

<Meta
	title='Related Resources'
	component={Paper}
	args={{
		data: {
			_id: '5f156b42577505d15e88f239',
			categories: {
				programmingLanguage: [],
			},
			tags: {
				features: ['Arbitrage'],
				topics: ['Blood'],
			},
			authors: [947228017269611],
			id: 18656838486747996,
			type: 'paper',
			name: 'This is a test paper',
			description: 'test paper description',
			activeflag: 'active',
			programmingLanguage: [],
			counter: 68,
			relatedresources: 5,
			persons: [
				{
					_id: '5e6f984a0a7300dc8f6fb196',
					id: 947228017269611,
					firstname: 'Test',
					lastname: 'Person',
				},
			],
			updated: 1634511600000,
		},
		onSearchPage: false,
		activeLink: false,
		showRelationshipQuestion: false,
	}}
/>

<Canvas>
	<Story name='Paper'>
		{props => {
			const updateOnFilterBadge = () => {
				alert('updateOnFilterBadge');
			};
			const removeButton = () => {
				alert('removed');
			};
			return <Paper {...props} updateOnFilterBadge={updateOnFilterBadge} removeButton={removeButton} />;
		}}
	</Story>
</Canvas>
