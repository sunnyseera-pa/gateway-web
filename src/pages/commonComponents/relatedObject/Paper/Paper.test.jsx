import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Paper from './Paper';
import mockData from './mockData';
const props = {
	data: { ...mockData },
	onSearchPage: false,
	activeLink: false,
	showRelationshipQuestion: false,
	updateOnFilterBadge: jest.fn(),
	removeButton: jest.fn(),
};
let wrapper;

describe('Given the Paper component', () => {
	describe('When it is rendered', () => {
		beforeAll(() => {
			wrapper = render(<Paper {...props} />);
		});

		it('Then matches the previous snapshot', () => {
			expect(wrapper.container).toMatchSnapshot();
		});

		it('Then Paper Title should be rendered with description', () => {
			expect(screen.getByTestId('paper-title')).toHaveTextContent(props.data.name);
			expect(screen.getByTestId('paper-description')).toHaveTextContent(props.data.description);
		});

		it('Then Paper SVG Icon should be rendered', () => {
			expect(screen.getByTestId('newprojecticon')).toBeTruthy();
		});

		it('Then the Features Badge  should be rendered without links', () => {
			props.data.tags.features.map(value => {
				expect(screen.getByTestId(`badge-${value}`)).toBeTruthy();
				expect(screen.queryByTestId(`badge-${value}-link`)).toBeNull();
			});
		});

		it('Then the Topic Badges should be rendered without links', () => {
			props.data.tags.topics.map(value => {
				expect(screen.getByTestId(`badge-${value}`)).toBeTruthy();
				expect(screen.queryByTestId(`badge-${value}-link`)).toBeNull();
			});
		});
	});

	describe('And activeLink is true', () => {
		it('Then the Tilte should be clickable with a link', () => {
			const { rerender } = wrapper;
			rerender(<Paper {...props} activeLink={true} />);
			expect(screen.getByTestId('paper-title')).toHaveAttribute('href', `/paper/${props.data.id}`);
		});
		it('Then the Features Badge/Tag should be rendered with links', () => {
			props.data.tags.features.map(value => {
				expect(screen.getByTestId(`badge-${value}`)).toBeTruthy();
				expect(screen.getByTestId(`badge-${value}-link`)).toHaveAttribute('href', `/search?search=&tab=Papers&paperfeatures=${value}`);
			});
		});

		it('Then the Topics Badge/Tag should be rendered with links', () => {
			props.data.tags.topics.map(value => {
				expect(screen.getByTestId(`badge-${value}`)).toBeTruthy();
				expect(screen.getByTestId(`badge-${value}-link`)).toHaveAttribute('href', `/search?search=&tab=Papers&papertopics=${value}`);
			});
		});

		describe('And onSearchPage is true', () => {
			let updateOnFilterBadge = jest.fn();
			it('Then Badge Tags/Features should be rendered without links', () => {
				const { rerender } = wrapper;
				rerender(<Paper {...props} activeLink={true} onSearchPage={true} updateOnFilterBadge={updateOnFilterBadge} />);
				props.data.tags.features.map(value => {
					expect(screen.getByTestId(`badge-${value}`)).toBeTruthy();
					expect(screen.queryByTestId(`badge-${value}-link`)).toBeNull();
				});
			});
			it('Then the Features Badge/Tag updateOnFilterBadge should be called', () => {
				fireEvent.click(screen.getByTestId(`badge-${props.data.tags.features[0]}`));
				expect(updateOnFilterBadge.mock.calls.length).toBe(1);
				expect(updateOnFilterBadge.mock.calls[0][0]).toEqual('paperFeaturesSelected');
				expect(updateOnFilterBadge.mock.calls[0][1]).toEqual({ label: props.data.tags.features[0], parentKey: 'paperfeatures' });
			});

			it('Then the Topic Badge/Tag updateOnFilterBadge should be called', () => {
				const { rerender } = wrapper;
				let updateOnFilterBadgeTopic = jest.fn();
				rerender(<Paper {...props} activeLink={true} onSearchPage={true} updateOnFilterBadge={updateOnFilterBadgeTopic} />);
				fireEvent.click(screen.getByTestId(`badge-${props.data.tags.topics[0]}`));
				expect(updateOnFilterBadgeTopic.mock.calls.length).toBe(1);
				expect(updateOnFilterBadgeTopic.mock.calls[0][0]).toEqual('paperTopicsSelected');
				expect(updateOnFilterBadgeTopic.mock.calls[0][1]).toEqual({ label: props.data.tags.topics[0], parentKey: 'papertopics' });
			});
		});
	});
	describe('And showRelationshipQuestion is true', () => {
		it('Then the remove button should be rendered ', () => {
			const { rerender } = wrapper;
			rerender(<Paper {...props} showRelationshipQuestion={true} />);
			expect(screen.getByTestId('closeicon')).toBeTruthy();
		});
		it('Then onclick removeButton function should be called', () => {
			fireEvent.click(screen.getByTestId('closeicon'));
			expect(props.removeButton.mock.calls.length).toBe(1);
		});
		it('Then the description should not be rendered', () => {
			expect(screen.queryByTestId('paper-description')).toBeNull();
		});
	});
});
