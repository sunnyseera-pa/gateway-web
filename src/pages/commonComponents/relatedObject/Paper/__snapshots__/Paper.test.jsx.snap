// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the Paper component When it is rendered Then matches the previous snapshot 1`] = `
<div>
  <div
    class="noMargin row"
    data-test-id="related-paper-object"
  >
    <div
      class="pad-left-24 col-lg-10 col-sm-10"
    >
      <span
        class="black-bold-16"
        data-testid="paper-title"
      >
         
        This is a test paper
      </span>
      <div
        class="gray800-14"
        style="margin-top: 2px;"
      />
      <div
        class="gray800-14"
        style="margin-top: 10px;"
      >
         
      </div>
    </div>
    <div
      class="pad-right-24 col-lg-2 col-sm-2"
    />
    <div
      class="pad-left-24 pad-right-24 pad-top-16 col-lg-12 col-sm-12"
    >
      <div
        class="badge-paper"
        data-testid="badge-Paper"
      >
        <svg
          class="badgeSvg mr-2"
          data-testid="newprojecticon"
          height="100%"
          viewBox="-2 -2 22 22"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <path
            d="M22,0 L22,17.4142136 L15.4142136,24 L2,24 L2,0 L22,0 Z M20,2 L4,2 L4,22 L14,22 L14,16 L20,16 L20,2 Z M12,16 L12,18 L6,18 L6,16 L12,16 Z M18,11 L18,13 L6,13 L6,11 L18,11 Z M18,6 L18,8 L6,8 L6,6 L18,6 Z"
            fill="#3c3c3b"
          />
        </svg>
        <span>
          Paper
        </span>
        <span />
      </div>
      <div
        class="badge-tag"
        data-testid="badge-Arbitrage"
      >
        <span>
          Arbitrage
        </span>
        <span />
      </div>
      <div
        class="badge-tag"
        data-testid="badge-Blood"
      >
        <span>
          Blood
        </span>
        <span />
      </div>
    </div>
    <div
      class="pad-left-24 pad-right-24 pad-top-24 pad-bottom-16 col-lg-12 col-sm-12"
    >
      <span
        class="gray800-14"
        data-testid="paper-description"
      >
        test paper description
      </span>
    </div>
  </div>
</div>
`;
